ddtrace/contrib/botocore/patch.py

```diff
@@ -467,10 +505,52 @@ def patched_api_call(original_func, instance, args, kwargs):
         span.set_tag(ANALYTICS_SAMPLE_RATE_KEY, config.botocore.get_analytics_sample_rate())
 
         try:
-            result = original_func(*args, **kwargs)
-            _set_response_metadata_tags(span, result)
+            if endpoint_name == "sqs" and operation == "ReceiveMessage" and config._data_streams_enabled:
+                queue_name = get_queue_name(params)
+
+                if "MessageAttributeNames" not in params:
+                    params.update({"MessageAttributeNames": ["_datadog"]})
+                elif "_datadog" not in params["MessageAttributeNames"]:
+                    params.update({"MessageAttributeNames": params["MessageAttributeNames"] + ["_datadog"]})
+
+                result = original_func(*args, **kwargs)
+                _set_response_metadata_tags(span, result)
+
+                try:
+                    if "Messages" in result:
+                        for message in result["Messages"]:
```

Comment:
nit: you could rewrite this if stack as a generator, which would make it more readable.

```
for message in messages_with_dsm_header(messages):
    ctx.foo()
```

You can get fancier, too, if you want to handle both matching and unmatching:

```
for message, has_dsm_header in _dsm_messages():
    if has_dsm_header:
        ctx.foo(message)
    else:
         ctx.new_path(message)
```